### PR TITLE
add docker files

### PR DESCRIPTION
**Pull request by saces**
simple alpine based docker container with `synapse_compress_state` and `synapse_auto_compressor`.

Example:

    docker run -it --rm -v /run/postgresql/.s.PGSQL.5432:/run/postgresql/.s.PGSQL.5432 registry.gitlab.com/mb-saces/rust-synapse-compress-state:testing synapse_auto_compressor -p "postgresql://db_user:db_password@%2Frun%2Fpostgresql/synapse" -n 100 -c 500

My work repos lives here: https://gitlab.com/mb-saces/rust-synapse-compress-state

### CODE REVIEW

**Comment by rlywtf** [comment]
@saces This Dockerfile builds the binary into a different location than is expected by https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/roles/matrix-synapse/defaults/main.yml#L616 which makes it a bit different than the existing available image here: https://hub.docker.com/r/devture/rust-synapse-compress-state.

image

**Comment by rlywtf** [comment]
> I'm going to keep poking around at things to see if I can get things working on my arm64 instance. 😅 Thank you for this PR!

I was able to adjust the final two COPY commands in this PR and then utilize a locally build arm64 docker image with the Ansible+Docker=Matrix project after overriding this variable. 🎉
- https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/roles/matrix-synapse/defaults/main.yml#L616 

`COPY --from=builder /opt/src/target/debug/synapse_compress_state /synapse-compress-state
COPY --from=builder /opt/src/target/debug/synapse_auto_compressor /synapse-auto-compressor`

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM rust:alpine AS builder
+
+RUN apk add python3 musl-dev pkgconfig openssl-dev make
+
+ENV RUSTFLAGS="-C target-feature=-crt-static"
+
+WORKDIR /opt/purginator3000/
```
Comment (squahtx) [inline Dockerfile:7]: Is this directory name arbitrary? Could we change it to something more appropriate?

**Review by squahtx — COMMENTED** [review]
Looks good, one question

**Comment by squahtx** [comment]
Could a sign-off be added, as per https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off ?

File: Dockerfile (hunk 1 above)
Comment (rlywtf) [reply Dockerfile:7]: @squahtx I used /opt/src/ in my local Dockerfile when testing [here ](https://github.com/matrix-org/rust-synapse-compress-state/pull/83#issuecomment-1038608156) and it worked fine. 😄

**Comment by rlywtf** [comment]
> Could a sign-off be added, as per https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off ?

I assume this question is for @saces as the submitter and not me? I'm happy to add mine if you would like it from reviewers as well. 😅 

Signed-off-by: @rlywtf <github@rly.wtf>

**Review by rlywtf — COMMENTED** [review]
I'm re-running my local test build now with these updates. _(on an arm64 instance)_

`> docker build . -t rust-synapse-compress-state | tee -a rust-synapse-compress-state_Dockerfile-build.log`

**Review by rlywtf — APPROVED** [review]

**Comment by squahtx** [comment]
> > Could a sign-off be added, as per https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off ?
> 
> I assume this question is for @saces as the submitter and not me?

That was for @saces, yes.

**Review by squahtx — APPROVED** [review]
lgtm!

**Pull request merged**